sphinx_gallery/gen_gallery.py

```diff
@@ -340,6 +340,33 @@ def call_memory(func):
     gallery_conf["compress_images"] = compress_images
     gallery_conf["compress_images_args"] = compress_images_args
 
+    # deal with matplotlib_animations
+    animations = gallery_conf["matplotlib_animations"]
+    if not isinstance(animations, (tuple, list)):
+        # Original option may be only a boolean.
+        animations = (animations,)
+    if len(animations) not in (1, 2):
+        raise ConfigError(
+            "Matplotlib animation options must be a single bool or "
+            f"(enabled: bool, format: str), not {animations!r}"
+        )
+    enabled = animations[0]
+    if not isinstance(enabled, bool):  # On/off boolean.
+        raise ConfigError("Animation enabled must be a bool")
```

Comment:
Is this too confusing?
```suggestion
    if len(animations) not in (1, 2) and not isinstance(enabled := animations[0], bool):
        raise ConfigError(
            "Matplotlib animation options must be a single bool or "
            f"(enabled: bool, format: str), not {animations!r}"
        )
```
This error message is informative for both errors and we save some config checking.

Reply:
If `animations = []`, then the second half of the condition will raise an `IndexError`.

Reply:
Python supports short circuiting, so if the first condition is false it doesn't bother evaluating the second

Reply:
There's a `not` there; the first condition is true for `[]`.

Reply:
Ah good point, my bad, didn't read properly. What about:
```python
if not (len(animations) in (1, 2) and isinstance(enabled := animations[0], bool)):
```
?

Reply:
Yes, I think that might work.